tools/vera++/scripts/rules/jerry_dereference_operator_always_on_right.tcl

```diff
@@ -18,7 +18,9 @@
 foreach f [getSourceFileNames] {
     set lineNumber 1
     foreach line [getAllLines $f] {
-        if {[regexp {\w\*\s\w+} $line]} {
+        if {[regexp {\w\*\s\w+} $line]
+             || [regexp {\w\*\)} $line]
+             || [regexp {\w\*$} $line]} {
```

Comment:
The first check is for `type* var_name`
The second is for type casting for a pointer type `(type*)`
The third one is for return values of functions. They are usually in a separate line.
The plus for `\w` is not necessary, we can remove if it bothers you. `\s+` may be a good idea.
About the name, my opinion is that the `*` symbol is the pointer dereference operator, but the expressions are really pointer declerations. I'd like to refer the `*` in the name of the rule not the expressions, but we can change the name of rule if it is misleading. Please suggest a proper rule name and message.